storage/innobase/trx/trx0purge.cc

```diff
@@ -495,14 +495,13 @@ inline dberr_t purge_sys_t::iterator::free_history_rseg(trx_rseg_t &rseg) const
 
 void purge_sys_t::cleanse_purge_queue(const fil_space_t &space)
 {
-  byte purge_elem_list[TRX_SYS_N_RSEGS];
+  purge_queue::container_type purge_elem_list;
   mysql_mutex_lock(&pq_mutex);
-  std::copy(purge_queue.c_begin(), purge_queue.c_end(), purge_elem_list);
-  byte *purge_list_end = purge_elem_list + purge_queue.size();
+  std::copy(purge_queue.cbegin(), purge_queue.cend(), purge_elem_list.begin());
   purge_queue.clear();
```

Comment:
Now that we have to use a variable-size heap-allocated container, could we just use the copy constructor?
```c++
purge_queue::container_type purge_elem_list(purge_queue.m_array);
```
I see that the data member is `private`, so this probably would not work directly as is. Could this be returned (with copy elision) by a member function?

Maybe, a cleaner way would be to make this a member function of `purge_queue` itself, just adding `mysql_mutex_assert_owner(&purge_sys.pq_mutex);` in the beginning? Everything except the mutex are private to `purge_queue`. Perhaps in that way, we could also remove the member functions `c_begin()` and `c_end()`?